src/charm.py

```diff
@@ -253,8 +253,14 @@ def _on_storage_attached(self, event: StorageAttachedEvent) -> None:
         if self.workload.active():
             self._set_status(Status.ADDED_STORAGE)
             self.workload.exec(f"chown -R {USER}:{GROUP} {self.workload.paths.data_path}")
-            self.workload.exec(f"chmod -R 770 {self.workload.paths.data_path}")
+            self.workload.exec(f"chmod -R 750 {self.workload.paths.data_path}")
             self._on_config_changed(event)
+        else:
```

Comment:
**question:** Do I understand correctly that the 'else' here is saying:
> If I'm not already active, it must be a re-attaching of an existing storage, handled at charm start-up
?

If I understood that right, I think we need to modify this to support re-attaching existing storage to a currently running unit, whereby in this example it won't work, right?

Maybe some modification which is like:
```python
if has_meta_properties(path=event.storage.path):  # no need for func, just example
```

Reply:
I reworked the condition in c0d2ac8, so that the execution path should be the same as before. If you are ok with this change, I'll clean up the commented code and merge on green.